lms/djangoapps/completion/api/v1/views.py

```diff
@@ -0,0 +1,80 @@
+""" API v1 views. """
+import logging
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import permissions
+
+from lms.djangoapps.completion.models import BlockCompletion
+from openedx.core.lib.api.permissions import IsStaffOrOwner
+
+log = logging.getLogger(__name__)
+
+
+class CompletionBatchView(APIView):
+    permission_classes = (permissions.IsAuthenticated, IsStaffOrOwner,)
+
+    def _validate(self, batch_object):
+        """
+        Performs validation on the batch object to make sure it is in the proper format.
+
+        Parameters:
+            batch_object: The object provided by the caller.
+
+        Raises:
+
+            KeyError:
+                If a key is not found in the object.
+
+            django.core.exceptions.ValidationError:
+                If a float is passed that is not between 0.0 and 1.0.
+        """
+        if 'user' not in batch_object:
+            raise KeyError("Key 'user' not found.")
+        if 'course_key' not in batch_object:
+            raise KeyError("Key 'course_key' not found.")
```

Comment:
I'd prefer to stick with "course_key."

edx-platform is pretty consistently inconsistent about this, as evidenced by:

```sh
~/work/opencraft/docker/edx-platform$ grep -r course_id lms cms openedx common|wc -l
8687
~/work/opencraft/docker/edx-platform$ grep -r course_key lms cms openedx common|wc -l 
7662
```

Within the completion API, we have consistently been using course_key and block_key so far, and I'd hate to start mixing them together.

Reply:
This may be a germane article: https://openedx.atlassian.net/wiki/spaces/OpenDev/pages/18350757/edX+REST+API+Conventions
`
Must: edX resource identifiers
Users should be referenced by username
Courses should be referenced via course keys
`

Reply:
No worries -- if the completion API calls them `course_key`, then `course_key` they can remain.